evm/src/fuzz/strategies/invariants.rs

```diff
@@ -0,0 +1,171 @@
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+use ethers::{
+    abi::{Abi, Function, ParamType},
+    types::{Address, Bytes},
+};
+use proptest::prelude::*;
+pub use proptest::test_runner::Config as FuzzConfig;
+
+use crate::fuzz::{
+    fuzz_calldata, fuzz_calldata_from_state,
+    invariant::{BasicTxDetails, FuzzRunIdentifiedContracts},
+    strategies::fuzz_param,
+    EvmFuzzState,
+};
+
+/// Given a target address, we generate random calldata.
+pub fn override_call_strat(
+    fuzz_state: EvmFuzzState,
+    contracts: FuzzRunIdentifiedContracts,
+    target: Arc<RwLock<Address>>,
+) -> SBoxedStrategy<(Address, Bytes)> {
+    let contracts_ref = contracts.clone();
+
+    let random_contract = any::<prop::sample::Selector>()
+        .prop_map(move |selector| *selector.select(contracts_ref.read().keys()));
```

Comment:
I recall @brockelmore saying that this was a very slow operation and that we should do it with a different way (or maybe we already do, I forget)?